util/topsql/reporter/datasink.go

```diff
@@ -23,14 +23,6 @@ type DataSink interface {
 	// the specified deadline, or the sink is closed, an error will be returned.
 	TrySend(data *ReportData, deadline time.Time) error
 
-	// IsPaused indicates that the DataSink is not expecting to receive records for now
-	// and may resume in the future.
-	IsPaused() bool
-
-	// IsDown indicates that the DataSink has been down and can be cleared.
-	// Note that: once a DataSink is down, it cannot go back to be up.
-	IsDown() bool
-
-	// Close cleans up resources owned by this DataSink
-	Close()
+	// CloseDataSink notifies DataSink that the reporter is dropping it.
+	CloseDataSink()
```

Comment:
Call it `Close` is enough? As it is already `DataSink::Close`.

Or maybe more precisely, `OnDeregisterFromReporter`? This seems to be more natural to me.